music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,332 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+# Name:         romanText/writeRoman.py
+# Purpose:      Writes Roman text files given a music21 stream of Roman numerals
+#
+# Authors:      Mark Gotham
+#
+# Copyright:    Copyright © 2020 Michael Scott Cuthbert and the music21 Project
+# License:      BSD, see license.txt
+# ------------------------------------------------------------------------------
+'''
+Writes Roman text files given a music21 stream of Roman numerals
+'''
+
+import fractions
+import os
+import unittest
+
+from music21 import exceptions21
+from music21 import environment
+environLocal = environment.Environment()
+
+
+# ------------------------------------------------------------------------------
+
+class WriteRomanException(exceptions21.Music21Exception):
+    pass
+
+
+# ------------------------------------------------------------------------------
+
+class RnWriter:
+    '''
+    Extracts the relevant information from a stream of Roman numeral objects for
+    writing to text files in the 'Roman text' format.
+
+    Includes handling of metadata such that entries for composer, title and more
+    can be user defined or extracted from score metadata wherever possible.
+    '''
+
+    def __init__(self,
+                 score,
+                 composer: str = '',
+                 title: str = '',
+                 analyst: str = '',
+                 proofreader: str = '',
+                 note: str = ''
+                 ):
+
+        self.score = score
+
+        self.romanNumerals = self.score.parts[0].recurse().getElementsByClass('RomanNumeral')
```

Comment:
I like that.

Reply:
Generally we make the initial writer be a general tool that if it's a non-Stream, we put it in a Measure, if it's a Measure (or generic Stream) we wrap it in a Part, and then if it's a Part, we wrap it in a Score, so that any music21 object can try to call "el.show('romanText')"  [btw -- does `show()` work, in addition to write?]